src/routing/approved.rs

```diff
@@ -102,9 +163,18 @@ impl Approved {
             event_tx,
             joins_allowed: true,
             resource_proof: ResourceProof::new(RESOURCE_PROOF_DATA_SIZE, RESOURCE_PROOF_DIFFICULTY),
+            end_users: EndUserRegistry::new(),
         }
     }
 
+    pub fn get_enduser_by_addr(&self, sender: &SocketAddr) -> Option<EndUser> {
+        self.end_users.get_enduser_by_addr(sender)
+    }
+
+    pub fn get_socket_addr(&self, id: &SocketId) -> Option<&SocketAddr> {
```

Comment:
better name the function to show it is only for enduser?

Reply:
Not sure how you mean.